lib/ui/views/settings/forumSettings/forum_settings_viewmodel.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:stacked/stacked.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class ForumSettingsViewModel extends BaseViewModel {
+  bool _isLoggedIn = false;
+  bool get isLoggedIn => _isLoggedIn;
+
+  void init() async {
+    _isLoggedIn = await checkLoggedIn();
+    notifyListeners();
+  }
+
+  void forumLogout() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+
+    prefs.setBool('loggedIn', false);
+    prefs.remove('username');
+    _isLoggedIn = false;
+    notifyListeners();
```

Comment:
I added these lines to remove the users `username` when they log out and to refresh the settings page so they don't still see the logout button. A `SnackBar`(similar to toasts) can be added in future PRs to inform the user that they have logged out.

Reply:
sounds like a plan!